### PR TITLE
Enable Java Docs.CI Onboarding

### PR DESCRIPTION
**Pull request by scbedd**
[Successful Test Release](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=483594&view=logs&s=d0889692-cae7-51e0-3e1f-b1d8c62a4117&j=df6dbc9d-9cb6-55ac-00c8-bc457d14297a)

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-java-release.yml [hunk 1]
```diff
@@ -98,9 +98,14 @@ stages:
                 deploy:
                   steps:
                     - checkout: self
+                    - template: /eng/pipelines/templates/steps/stage-artifacts.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-java-release.yml:101]: I think this has been removed now with @mitchdenny's recent work.

Comment (scbedd) [reply eng/pipelines/templates/stages/archetype-java-release.yml:101]: You're absolutely correct 👍

Comment (scbedd) [reply eng/pipelines/templates/stages/archetype-java-release.yml:101]: Updated.

File: eng/pipelines/templates/stages/archetype-java-release.yml [hunk 2]
```diff
@@ -152,14 +152,19 @@ stages:
               environment: github
               dependsOn: PublishPackage
               variables:
-                - template: ../variables/globals.yml
+                CIConfigs: "{'targets':[{'path_to_config':'package.json','mode':'Preview','monikerid':'1'},{'path_to_config':'package.json','mode':'Latest','monikerid':'0'}]}"
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-java-release.yml:155]: This CIConfig is a little hard to read/parse inline in the yml. Do we have any better options to pass this data? Perhaps we could also put it as a config file that we read.

Comment (scbedd) [reply eng/pipelines/templates/stages/archetype-java-release.yml:155]: Yeah. I'll shift it to a config file location in the repo in another PR.

**Pull request merged**